src/serial/serial.c

```diff
@@ -59,27 +63,52 @@ struct Serial {
 	struct serial_cfg cfg;
 };
 
+void serial_purge_rx_queue(struct Serial* s)
+{
+	xQueueReset(s->rx_queue);
+}
+
+void serial_purge_tx_queue(struct Serial* s)
+{
+	xQueueReset(s->tx_queue);
+}
+
+/**
+ * Clears the contents of the rx and tx queues.
+ */
+void serial_clear(struct Serial *s)
+{
+        serial_purge_rx_queue(s);
+        serial_purge_tx_queue(s);
+}
+
+/**
+ * Unblocks our Seral rx queue by putting an invalid character on the
+ * front of the queue. This will unblock any task that is waiting on
+ * the queue.  Its up to the rx handler below to ensure that we return
+ * the correct status code.
+ */
+static void unblock_rx_queue(struct Serial *s)
+{
+	xQueueSendToFront(s->rx_queue, &invalid_char, 0);
```

Comment:
So we're using the invalid char as a flag  that says "close this queue"? Seems legit.